spark/src/main/scala/org/apache/spark/sql/comet/execution/shuffle/CometShuffleExchangeExec.scala

```diff
@@ -208,6 +216,50 @@ object CometShuffleExchangeExec extends ShimCometShuffleExchangeExec {
     dependency
   }
 
+  /**
+   * This is copied from Spark `ShuffleExchangeExec.needToCopyObjectsBeforeShuffle`. The only
+   * difference is that we use `BosonShuffleManager` instead of `SortShuffleManager`.
+   */
+  private def needToCopyObjectsBeforeShuffle(partitioner: Partitioner): Boolean = {
```

Comment:
Spark `needToCopyObjectsBeforeShuffle` will return `true` if it finds the shuffle manager is not Spark `SortShuffleManager`. So it incurs additional row copying for all cases which are redundant for Comet and increase memory usage.

Reply:
This could also improve the shuffle time on columnar shuffle as it removes expensive copying operation.

Reply:
So not a memory leak, just double the memory usage?

Reply:
Yea, I think so.

Reply:
[a bit orthogonal] how did you find it ? by profiling or by just analyzing the shuffle size, what was the scale of tpcds (3TB) curious ?

Reply:
I did memory profiling when running TPCDS queries locally. Because it is local benchmark, the scale is small. This issue is not only for larger scale so it can be profiled in small scale too.